web_infrastructure/jenkins_job.py

```diff
@@ -0,0 +1,332 @@
+        changed = False
+        if self.job_exists(module):
+            changed = True
+            if not module.check_mode:
+                try:
+                    self.server.delete_job(self.name)
+                except Exception:
+                    e = get_exception()
+                    module.fail_json(msg='Unable to delete job, %s for %s' % (str(e), self.jenkins_url))
+
+        module.exit_json(changed=changed, name=self.name, state=self.state, url=self.jenkins_url)
+
+    def disable_job(self, module):
+        changed = False
+        if self.job_exists(module):
+            status = self.get_job_status(module)
+            try:
+                if status != "disabled":
+                    changed = True
+                    if not module.check_mode:
+                        self.server.disable_job(self.name)
+            except Exception:
+                e = get_exception()
+                module.fail_json(msg='Unable to disable job, %s for %s' % (str(e), self.jenkins_url))
+
+        module.exit_json(changed=changed, name=self.name, state=self.state, url=self.jenkins_url)
+
+def test_dependencies(module):
+    if not python_jenkins_installed:
+        module.fail_json(msg="python-jenkins required for this module. "\
+              "see http://python-jenkins.readthedocs.io/en/latest/install.html")
+
+    if not python_lxml_installed:
+        module.fail_json(msg="lxml required for this module. "\
+              "see http://lxml.de/installation.html")
+
+def job_config_to_string(xml_str):
+    return ET.tostring(ET.fromstring(xml_str))
+
+def xml_to_string(source):
+    return ET.tostring(ET.parse(source).getroot())
+
+def jenkins_builder(module):
+    if module.params.get('name') and module.params.get('config_file') and module.params.get('url'):
+        return Jenkins(
+            module.params.get('config_file'),
+            module.params.get('name'),
+            module.params.get('password'),
+            module.params.get('state'),
+            module.params.get('token'),
+            module.params.get('url'),
+            module.params.get('user')
+        )
+    else:
+        module.fail_json(msg='name, state, url are required: name=%s, config_file=%s, url=%s' % (
+                            module.params.get('name'),
+                            module.params.get('config_file'),
+                            module.params.get('url'),
+                            str(e)))
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            config_file = dict(required=True, type='path'),
+            name        = dict(required=True),
+            password    = dict(required=False, no_log=True),
+            state       = dict(required=False, default='present', choices=['present', 'absent', 'disabled']),
+            token       = dict(required=False, no_log=True),
+            url         = dict(required=False, default="http://localhost:8080"),
+            user        = dict(required=False)
```

Comment:
You can have a Jenkins server with no authentication and because of that reason I don't think it should have default value either. Regarding the import conditions mentioned in previous comments, let me remove the unusued lines.